### PR TITLE
iOS SDK updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added steps for SPM installation in both the v1 and v2 docs.
> The doc now dynamically loads the SDK version in the shields badge.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx [hunk 1]
```diff
@@ -137,11 +134,72 @@ import Rudder
...
+    <Tab>Xcode</Tab>
+    <Tab>Swift</Tab>
+  </TabList>
+    <TabPanels>
+      <TabPanel>
+        To add the RudderStack package in Xcode, follow these steps:
+<ol>
+  <li>Go to <strong>File</strong> - <strong>Add Package</strong>.</li>
+  <li>In the search bar, enter the package repository <code class="inline-code">git@github.com:rudderlabs/rudder-sdk-ios.git</code>.</li>
+  <li>In <strong>Dependency Rule</strong>, select  <strong>Up to Next Major Version</strong> and enter the value as <strong>1.8.0</strong>, as shown:</li>
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx:155]: Can you fetch this version from `package.json`?

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx [hunk 2]
```diff
@@ -137,11 +134,72 @@ import Rudder
...
+    name: "RudderStack",
+    products: [
+        // Products define the executables and libraries a package produces, and make them visible to other packages.
+        .library(
+            name: "RudderStack",
+            targets: ["RudderStack"]),
+    ],
+    dependencies: [
+        // Dependencies declare other packages that this package depends on.
+        .package(url: "git@github.com:rudderlabs/rudder-sdk-ios.git", from: "1.8.0")
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx:182]: Can you fetch this version from `package.json`?

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/ios-v2.mdx [hunk 3]
```diff
@@ -136,6 +134,69 @@ import Rudder
...
+    <Tab>Xcode</Tab>
+    <Tab>Swift</Tab>
+  </TabList>
+    <TabPanels>
+      <TabPanel>
+        To add the RudderStack package in Xcode, follow these steps:
+<ol>
+  <li>Go to <strong>File</strong> - <strong>Add Package</strong>.</li>
+  <li>In the search bar, enter the package repository <code class="inline-code">git@github.com:rudderlabs/rudder-sdk-ios.git</code>.</li>
+  <li>In <strong>Dependency Rule</strong>, select  <strong>Up to Next Major Version</strong> and enter the value as <strong>2.2.5</strong>, as shown:</li>
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ios-sdk/ios-v2.mdx:152]: Can you fetch this version from `package.json`?

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/ios-v2.mdx [hunk 4]
```diff
@@ -136,6 +134,69 @@ import Rudder
...
+    name: "RudderStack",
+    products: [
+        // Products define the executables and libraries a package produces, and make them visible to other packages.
+        .library(
+            name: "RudderStack",
+            targets: ["RudderStack"]),
+    ],
+    dependencies: [
+        // Dependencies declare other packages that this package depends on.
+        .package(url: "git@github.com:rudderlabs/rudder-sdk-ios.git", from: "1.8.0")
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ios-sdk/ios-v2.mdx:179]: Can you fetch this version from `package.json`?

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx:155]: I'm not sure how to do that, @pallabmaiti - could you or @1abhishekpandey please suggest the relevant steps/snippet to fetch the version automatically?

**Review by pallabmaiti — APPROVED** [review]

File: docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx (hunk 1 above)
Comment (pallabmaiti) [reply docs/sources/event-streams/sdks/rudderstack-ios-sdk/index.mdx:155]: Got it.

**Review by 1abhishekpandey — APPROVED** [review]

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**